### DISCUSSION TITLE
What to edit when translating to new locale

### DISCUSSION THREAD

**Discussion by thoni56** [alan-if/alan-i18n#2] [New Locales]
I started on the Swedish challenge, that I got.
I'm posting here to announce two things that I almost immediately decided to do which is not exactly translating.

1. I removed all historical version history, primarily since it's so old and probably not interesting any more, but also because it only applies to the original (English) library.
2. I improved the responses from something like `"You don't have that."` to using what now feels like the natural `"You don't have" say the obj. "."`

In line with #1 we probably also should clean out the various README and other administrative documents from the translations assuming they are still in the original.

Comments on those decisions are welcome.

**Comment by tajmone** [comment]
I agree on both points, and we should probably do some clean-up operations also in the Spanish library and the English one — I postponed those because I wanted to (1) leave the original files in the repo history, (2) be sure I won't remove any useful info for future maintainers.

We should probably have a look at the English library source and tweak deprecated syntaxes (e.g. `$o`) in favour of idiomatic ALAN 3 constructs.

# New Locale Guidelines

I also have a few proposal for anyone starting to work on a new locale:

1. Temporary development should be done in a dev branch, until ready to merge in the main repository. The branch should be named according to the new locale, using a two-letter country code, e.g. `dev-it`, etc.
2. The new locale should be in a folder named as `alan_XX` where `XX` stands for the two-letter [ISO 639-1] code representing the language (e.g. `alan_it` for Italian).
3. The main `README.md` inside the locale folder should be in English, for the benefit of maintainers and end users (copyright, license, etc.), and a second README file in the target locale should be created in the form `README_XX.md` (e.g. `README_IT.md` for Italian).
4. The actual library should be named as `lib_XX`, where `XX` again stands for the locale two-letter code.
5. Ideally, each library should contain a sample adventure (e.g. _Cloak of Darkness_, or something similar) with its solution file to allow testing its compilation and execution.

These points should be added to the `CONTRIBUTING.md` document, once we've agreed on the guidelines.

Some questions are not covered by the above:

- What if different variations of a same locale are present? e.g. `EN_US` and `EN_UK` (for USA and UK English, respectively).
    + Should they be both inside `alan_en`? (e.g. `alan_en/lib_en_uk`, etc.)
    + Or should we create independent root folders? (e.g. `alan_en_uk` and `alan_en_us`)

It's quite possible that end users might create these library variations of a same locale, after all they only require some spelling tweaking here and there, and we've seen this happen with many other IF systems, especially when authors want to create classical IF work (e.g. Gothic, etc.).



[ISO 639-1]: https://en.wikipedia.org/wiki/ISO_639-1

**Reply by thoni56** [reply]
Those are good guidelines.

A set of translated _Cloak of Darkness_, one for each translation, would be cool. We can make that a part of the translation process/instructions.

Re: variations, let's cross that when we get to it, but root folders would be my suggestion. So if we have `alan_sv` (which is probably actually `alan_sv_se`) and someone makes the `alan_sv_fi` translation, then add that new root and we'll rename `alan_sv` as appropriate.
